### DISCUSSION TITLE
HTTP Enrichment

### DISCUSSION THREAD

**Discussion by owen5mith** [vectordotdev/vector#14039] [Q&A] [answered]
Hello,

I'm trying to understand if Vector will support my use case.

I have a source of log events that all contain a UUID. I'm looking to merge these with another JSON object that can only be accessed by a HTTP API, for example, at  http://site.com/<UUID>/data.

I understand this should be possible with the Lua transform, however I wanted to check if anyone had an idea of how this could be done without Lua to aid the performance of running this over every event.

One idea I've had is using the new CSV Enrichment. A script next to vector could refresh the CSV script by calling http://site.com/all/data, and converting this into a CSV every minute, however I'm unsure if the enrichment is reloaded from disk after vector has first started up?

**Comment by jszwedko** [comment, marked as answer]
Hi @owen5mith !

Currently the `lua` transform is the best option here. We do have an open issue for an `http` transform that I think would cover this use case: https://github.com/vectordotdev/vector/issues/1385 . You can track there for any updates.
